### DISCUSSION TITLE
@typegoose/typegoose@7.6.1 has incorrect peer dependency mongoose@5.10.0 - 5.10.18

### DISCUSSION THREAD

**Discussion by Juan-hwt** [typegoose/typegoose#558] [Q&A] [answered]
## Versions

- System: linux / macos / windows 
- NodeJS: 0.0.0
- Typescript: 0.0.0

- Compiler / Transpiler: tsc / ts-node / ts-jest 
- Typegoose(NPM): 0.0.0 
- Typegoose(GIT): commithash
- mongoose: 0.0.0 
- mongodb: 0.0.0
- nestjs / nestjs-typegoose: 0.0.0 / 0.0.0 
- grahpql / type-graphql: 0.0.0 / 0.0.0 

---

run:
`yarn add mongoose @types/mongoose @typegoose/typegoose nestjs-typegoose`
give me this:
```
[3/4] Linking dependencies...
warning " > @typegoose/typegoose@7.6.1" has incorrect peer dependency "mongoose@5.10.0 - 5.10.18".
```
It can run, but how can I fix the problem? thank you

**Comment by smolinari** [comment]
Follow the instructions (and order of installation):

> npm i -s @typegoose/typegoose # install typegoose itself
> npm i -s mongoose # install peer-dependency mongoose
> npm i -D @types/mongoose # install all types for mongoose - this is required for typegoose to work in TypeScript

Scott

**Reply by hasezoey** [reply]
actually the order does not matter (especially when installed in one command)

**Reply by smolinari** [reply]
And, that was what I thought, until I looked at the docs and thought, "oh, npm or yarn will take into account peer dependencies." because of the comment behind the mongoose install (`install peer-dependency mongoose`). But, I guess it means, "be sure to install the peer-dependency mongoose" or better (`npm i -s mongoose@5.10.18`). 😁  

Scott

**Reply by smolinari** [reply]
Yeah, sometimes I'm a noob like that too. 😊 

Scott

**Reply by hasezoey** [reply]
npm (at least in 6.x) did not install peer dependencies automatically (i think some version before did)
and i *think* yarn also does not (but i am not sure)

**Comment by hasezoey** [comment, marked as answer]
sorry if this was not clear in the documentation, typegoose 7.x highest supported mongoose version is `5.10.18`, but by default without specifying an version, the latest will be installed, so modify your command to:
`yarn add mongoose@5.10.18 @types/mongoose @typegoose/typegoose nestjs-typegoose`
